addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5874,3 +5874,58 @@ def test_partial_reconcile_amounts_of_several_matching_lines(self):
             {'amount': 1001.0, 'debit_move_id': line_2.id, 'credit_move_id': line_5.id},
             {'amount': 1002.0, 'debit_move_id': line_3.id, 'credit_move_id': line_4.id},
         ])
+
+    def test_exchange_move_assignment_with_group_payment(self):
+        """
+        Test that when doing a group payment the exchange_moves are correctly assigned to the invoices
+        """
+        foreign_curr = self.env['res.currency'].create({
+            'name': "Sushi",
+            'symbol': '🍣',
+            'rounding': 0.01,
+            'rate_ids': [
+                Command.create({'name': '2025-01-01', 'rate': 0.054493834023}),
+                Command.create({'name': '2025-01-02', 'rate': 0.054363189597}),
+            ],
+        })
+
+        inv1 = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'invoice_date': '2025-01-01',
+            'partner_id': self.partner_a.id,
+            'currency_id': foreign_curr.id,
+            'invoice_line_ids': [Command.create({
+                'product_id': self.product_a.id,
+                'price_unit': 500.0,
+                'tax_ids': [],
+            })],
+        })
+        inv1.action_post()
+
+        inv2 = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'invoice_date': '2025-01-01',
+            'partner_id': self.partner_a.id,
+            'currency_id': foreign_curr.id,
+            'invoice_line_ids': [Command.create({
+                'product_id': self.product_a.id,
+                'price_unit': 10.0,
+                'tax_ids': [],
+            })],
+        })
+        inv2.action_post()
+
+        active_ids = (inv1 + inv2).ids
+        pmt_wizard = self.env['account.payment.register'].with_context(active_model='account.move', active_ids=active_ids).create({
+            'payment_date': '2025-01-02',
+            'group_payment': True,
+            'amount': 510.0,
+            'currency_id': foreign_curr.id
+        })
+        pmt_wizard._create_payments()
+        payment = self.env['account.payment'].search([])
```

Comment:
To check but I think you can do that
```suggestion
        payment = self.env['account.payment.register'].with_context(active_model='account.move', active_ids=active_ids).create({
            'payment_date': '2025-01-02',
            'group_payment': True,
            'amount': 510.0,
            'currency_id': foreign_curr.id
        })._create_payments()
```